### ISSUE TITLE
todo does not provide visit/depart_todolist

### ISSUE THREAD

**Issue by papadeltasierra** [sphinx-doc/sphinx#4899]
Subject: Unable to add todolist formatting that is distinct from individual todos

### Problem
- The todolist is actually output as one or more todo/paragraph pairs with no wrapper node that can be identified to allow the addition of distinct formatting for todo lists.

#### Procedure to reproduce the problem
```
Any command that builds a todo and list - the output is OK but cannot be customized.
```

#### Error logs / results
```
None appropriate.
```

#### Expected results
I would like to be able to do things like:
- An outer wrapper for the ToDo list
- A list of numbered entries, each of which is
  - The todo text (todo node, ideally without the 'doto:' prefix)
  - The 'original todo' link sentence (paragraph node)

### Reproducible project / your project
Any.

### Environment info
- Linux
- Python 2.7
- Sphinx version: 1.7.2
- Actually using Confluence plugin but HTML produces similar results.

**Comment by tk0miya** [comment]
Yes, `todolist` node is replaced with a list of `todo` nodes on resolving phase. So visitor functions on writer are never called.

+0: wrapping whole of todo list with `<div>` is reasonable
-0: arranging todo label and link sentence is too much. Using extension is better for such case.
